client.go

```diff
@@ -17,11 +19,12 @@ var _ error = StatusError{}
 type StatusError struct {
 	Status     string
 	StatusCode int
+	RetryAfter time.Duration
```

Comment:
I thought about the type of the RetryAfter when making this change in the Hauser project, and had decided to just propagate the header as string. That way the end client doesn't have to decide to interpret a 0 duration as an incorrect header value or an actual 0 duration retry-after directive. Maybe I'm overthinking this, but just thought I would share my thoughts.

Reply:
Good point. In Go, this is typically handled by documentation. Would something like the following help?

> StatusCode field represents the status code on the response. If the code is http.StatusTooManyRequests, the RetryAfter field is set indicating the duration to wait before retrying the operation.

Reply:
Retry after can also be set in a few other responses (503 for example). Maybe we explicitly say that if the retry-after duration is 0 seconds in the status error that could mean either the retry-after was not applicable to the response or the header was malformed. The end user should make decisions based on the combination of status code and retry after field?

Reply:
Sorry it's taken me so long to come back to this, had a few complex tickets this week. My intent was always to let the client make decisions based on both `StatusCode` and the `RetryAfter` fields, but we can't ultimately control clients' behavior here. In the corresponding hauser PR, I hope those 2 fields have been used well!

About how to represent the `Retry-After` header, I did argue with myself a bit, and this was kind of my final set of thoughts around it:
 * None of our APIs ever return Retry-After as a date, and our reverse proxy behaves the same way when it generates a 503. It's safe to consistently interpret this header as an integer representing seconds to wait, and I think `time.Duration` is the most fitting way to represent that information in go.
 * Checking for a zero-value string (`""`) is not much different than checking for a zero value duration. You still have to write an if statement with comparison to a constant value. And if the string has already been converted to a duration, you don't have to additionally check for the case where it's a non-empty string with some invalid value.
 * The most common use case for the `Retry-After` header is to go ahead and wait for the duration contained therein, and to do that you'd have to convert to a `time.Duration` value. It seems better to keep that work centralized, rather than to repeat it in all the clients.

That said, how about the following slight modification to Nishanth's original doc comment suggestion (because I think `Retry-After` can also be sent on a 301 response). Doc for `StatusCode`:

>StatusCode field represents the HTTP status code on the response. 

Doc for `RetryAfter`:
>RetryAfter field is set from the `Retry-After` header, when it is available on the response. When the header is missing, empty, or is not a parsable as an integer, RetryAfter will be a duration of length 0.

Reply:
I agree with Jess on this one. It's better to have `RetryAfter`'s field as `time.Duration` and it makes sense to parse it as soon as possible so that all of the clients do not have to do the same conversion. Also, the doc for the `RetryAfter` field looks good.